dotCMS/src/main/java/com/dotcms/graphql/business/GraphqlAPIImpl.java

```diff
@@ -160,137 +99,37 @@ private void printSchema() {
-
-        graphqlObjectTypes.put(graphQLType.getName(), graphQLType);
-    }
-
-    private void handleRelationshipField(final ContentType contentType, GraphQLObjectType.Builder builder,
-                                         final Field field, final Map<String, GraphQLObjectType> typesMap) {
-
-        final ContentType relatedContentType;
-        try {
-            relatedContentType = getRelatedContentTypeForField(field, APILocator.systemUser());
-        } catch (DotSecurityException | DotDataException e) {
-            throw new DotStateException("Unable to create relationship field type for field: " + contentType.variable() + "." + field.variable(), e);
         }
 
-        Relationship relationship;
+        List<GraphQLFieldDefinition> fieldDefinitions = new ArrayList<>();
 
-        try {
-            relationship = relationshipAPI.getRelationshipFromField(field,
-                APILocator.systemUser());
-        } catch (DotDataException | DotSecurityException e) {
-            throw new DotRuntimeException(e);
+        for (GraphQLFieldsProvider fieldsProvider : fieldsProviders) {
+            try {
+                fieldDefinitions.addAll(fieldsProvider.getFields());
+            } catch (DotDataException e) {
+                Logger.error("Unable to get types for type provider:" + fieldsProvider.getClass(), e);
+            }
         }
 
-        final ContentletRelationships contentletRelationships = new ContentletRelationships(null);
-        final ContentletRelationships.ContentletRelationshipRecords
-            records = contentletRelationships.new ContentletRelationshipRecords(
-            relationship,
-            relationshipAPI.isChildField(relationship, field));
-
-        GraphQLOutputType outputType = typesMap.get(relatedContentType.variable()) != null
-            ? typesMap.get(relatedContentType.variable())
-            : GraphQLTypeReference.typeRef(relatedContentType.variable());
-
-
-        outputType = records.doesAllowOnlyOne()
-            ? outputType
-            : list(outputType);
-
-        builder.field(newFieldDefinition()
-            .name(field.variable())
-            .type(field.required()?nonNull(outputType):outputType)
-            .dataFetcher(new RelationshipFieldDataFetcher())
-        );
-    }
-
-    @Override
-    public GraphQLOutputType getGraphqlTypeForFieldClass(final Class<? extends Field> fieldClass, final Field field) {
-        return fieldClassGraphqlTypeMap.get(fieldClass)!= null
-            ? fieldClassGraphqlTypeMap.get(fieldClass)
-            : fieldClass.equals(TextField.class) && field.dataType().equals(DataTypes.INTEGER) ? GraphQLInt
-                : fieldClass.equals(TextField.class) && field.dataType().equals(DataTypes.FLOAT) ? GraphQLFloat
-                    : GraphQLString;
-    }
+        // Root Type
+        GraphQLObjectType.Builder rootTypeBuilder = createRootTypeBuilder().fields(fieldDefinitions);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'rootTypeBuilder' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642169)